src/magnetVacuumFF.cpp

```diff
@@ -242,34 +242,37 @@ static Ref_t create_detector(Detector& det, xml_h e, SensitiveDetector /* sens *
 
         if(makeIP_B0pfVacuum){
 
-                double specialGapLength = sqrt(pow(z_beg[0] - endOfCentralBeamPipe_z, 2) + pow(x_beg[0] - endOfCentralBeamPipe_x, 2)) - 0.1;
-                        double specialGap_z = 0.5*specialGapLength*cos(crossingAngle) + endOfCentralBeamPipe_z;
-                        double specialGap_x = 0.5*specialGapLength*sin(crossingAngle) + endOfCentralBeamPipe_x;
+        double specialGapLength = sqrt(pow(z_beg[0] - endOfCentralBeamPipe_z, 2) + pow(x_beg[0] - endOfCentralBeamPipe_x, 2)) - 0.1;
+        double specialGap_z = 0.5*specialGapLength*cos(crossingAngle) + endOfCentralBeamPipe_z;
+        double specialGap_x = 0.5*specialGapLength*sin(crossingAngle) + endOfCentralBeamPipe_x;
 
-                        std::string piece_name  = Form("GapVacuum%d", numGaps + numMagnets);
+        std::string piece_name  = Form("GapVacuum%d", numGaps + numMagnets);
 
-                        Cone specialGap(piece_name, specialGapLength/2, 0.0, vacuumDiameterEntrance/2, 0.0, vacuumDiameterExit/2 );
+        Cone specialGap(piece_name, specialGapLength/2, 0.0, vacuumDiameterEntrance/2, 0.0, vacuumDiameterExit/2 );
 
-                        Volume specialGap_v(piece_name, specialGap, m_Vac);
-                        sdet.setAttributes(det, specialGap_v, x_det.regionStr(), x_det.limitsStr(), vis_name);
+                Volume specialGap_v(piece_name, specialGap, m_Vac);
+                sdet.setAttributes(det, specialGap_v, x_det.regionStr(), x_det.limitsStr(), vis_name);
 
-                        auto pv = assembly.placeVolume(specialGap_v, Transform3D(RotationY(crossingAngle), Position(specialGap_x, 0.0, specialGap_z)));
-                        pv.addPhysVolID("sector", 1);
+                auto pv = assembly.placeVolume(specialGap_v, Transform3D(RotationY(crossingAngle), Position(specialGap_x, 0.0, specialGap_z)));
+                pv.addPhysVolID("sector", 1);
 
-                        DetElement * tmp = new DetElement(sdet, Form("sector%d_de", numGaps + numMagnets), 1);
-                        detectorElement.push_back(tmp);
+                DetElement tmp(sdet, Form("sector%d_de", numGaps + numMagnets), 1);
+                detectorElement.push_back(tmp);
 
-                        detectorElement[numGaps + numMagnets]->setPlacement(pv);
+                detectorElement[numGaps + numMagnets].setPlacement(pv);
```

Comment:
The code always accesses the last element of the detector, so the vector itself seems not necessary.
The following code should've done the same job.
`DetElement tmp(sdet, Form("sector%d_de", numGaps + numMagnets), 1);`
'tmp.setPlacement(pv);`